xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -2463,6 +2471,7 @@ void CVideoPlayer::OnExit()
   bookmark.timeInSeconds = GetTime() / 1000;
   bookmark.player = m_name;
   bookmark.playerState = GetPlayerState();
+  fileItem.SetStreamDetailsFromVideoPlayer(videoInfo, bookmark.totalTimeInSeconds, audioInfo, subtitleInfo);
```

Comment:
Did you mean this? I've removed all specifics from VP code. Just Get*StreamInfo left and call to CFileItem function.

Maybe I misunderstood you, but we don't need to make VP to do all work with CFileItem updating and call any VP methods from outside, we need just to pass stream details inside fileItem to OnPlayerCloseFile(fileItem, bookmark) cb. The same as bookmark.
1. VP job: CVideoPlayer::OnExit() - when playback has stopped
    * [new code] before closing streams VP gets current stream info
    * [existing code] create a copy of m_item as fileItem
    * [new code] pass stream details to fileItem by calling fileItem.SetStreamDetailsFromVideoPlayer(...)
    * [existing code] submit OnPlayerCloseFile cb with our fileItem, containing stream details 
2. Application job: CApplication::OnPlayerCloseFile() - save stream details from VP's fileItem to database

Or should I completely remove all new code from VP?

Reply:
> Did you mean this? 

No. Now a very specific behaviour is distributed over serveral components. It should be obvious where the request comes from and how it is executed. There should dbe no logical from CFileItem to VP.

> Or should I completely remove all new code from VP?

No. Implement a new method on VP that tells VP that it should update fileItem with streams on exit.

Reply:
Do you mean something like this?
```
CVideoPlayer::SetIfNeededToUpdateFileItemStreamDetais(bool set_details)
{
  m_setStreamDetailsToFileItem = set_details;
}

CVideoPlayer::OnExit()
{
  if (m_setStreamDetailsToFileItem)
  {
    // get current stream info
  }
  ...
  if (m_setStreamDetailsToFileItem)
    fileItem.SetStreamDetailsFromVideoPlayer(...);
  m_outboundEvents->Submit([=]() { cb->OnPlayerCloseFile(fileItem, bookmark); });
}
```
May I check it inside VP?
```
CVideoPlayer::OpenFile(...)
{
  if (m_item.IsNeededToSetStreamDetailsFromVideoPlayer())
    m_setStreamDetailsToFileItem = true;
}
```
Because if I should call this method from outside, then I have to add this method to IVideoPlayer interface too, because players created with factory... or check with dynamic_cast if created player is CVideoPlayer...